integrations/spark/spark-job-dataquality/spark-job-dataquality-app/src/main/java/com/thinkbiganalytics/spark/dataquality/checker/DataQualityChecker.java

```diff
@@ -20,96 +20,398 @@
+
+import com.thinkbiganalytics.spark.dataquality.rule.DataQualityRule;
+import com.thinkbiganalytics.spark.dataquality.rule.InvalidRowPercentRuleImpl;
+import com.thinkbiganalytics.spark.dataquality.rule.InvalidRowTotalCountRuleImpl;
+import com.thinkbiganalytics.spark.dataquality.rule.RowCountRuleImpl;
+import com.thinkbiganalytics.spark.dataquality.rule.SourceToFeedCountRuleImpl;
+
 import org.apache.commons.lang3.StringUtils;
 import org.apache.spark.SparkContext;
-import org.apache.spark.sql.Row;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.sql.DataFrame;
 import org.apache.spark.sql.hive.HiveContext;
+import org.json.simple.JSONObject;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.ApplicationContext;
 import org.springframework.context.annotation.AnnotationConfigApplicationContext;
 import org.springframework.stereotype.Component;
 
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 /**
- * Perform data quality checks
+ * This class performs the Data Quality Checks. This is done by executing various Data Quality rules
+ * and using the passed in JSON file which contains all the necessary attributes. The results of all
+ * the tests are stored into a feed_dataquality hive table<br>
+ * <br>
+ * This class takes in one argument which is the path of the JSON file which contains all the
+ * flowfile attributes<br>
+ * <br>
+ * Failure is considered to be if any of the rules failed.<br>
+ * <br>
+ * Please refer to README for commands to run application.<br>
  */
-
-/*
-    TODO: Implement full functionality
-
-    This implementation provides the skeleton layout to enable implementation of the full functionality.
-     It demonstrates using Spark to run a row count for a schema.table in Hive.
-     Tested on both Spark 1 and Spark 2.
-     Please refer to README for commands to run application.
- */
-
 @Component
 public class DataQualityChecker {
-
     private static final Logger log = LoggerFactory.getLogger(DataQualityChecker.class);
 
     @Autowired
     private SparkContextService scs;
-
     private HiveContext hiveContext;
-    private String categoryName;
-    private String feedName;
 
-    public static void main(String[] args) {
+    private FlowAttributes flowAttributes;
+    private Map<String, DataQualityRule> availableRules;
+    private List<DataQualityRule> ruleList;
 
+    public static void main(String[] args) {
         log.info("Running DataQualityChecker with these command line args: " + StringUtils.join(args, ","));
 
-        if (args.length < 2) {
-            System.out.println("Expected command line args: <hive-schema-name> <hive-table-name>");
+        if (args.length < 1) {
+            System.out.println("Expected command line args: <path-to-attribute-file>");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS106)